app/controllers/api/v1/search_controller.rb

```diff
@@ -664,32 +664,65 @@ def self.match_results_by_filter(search_result:, result_key:, facets:)
       end
 
       # execute a search in TDR and get back normalized results
-      def self.get_tdr_results(selected_facets)
-        query_json = ApplicationController.data_repo_client.generate_query_from_facets(selected_facets)
-        raw_tdr_results = ApplicationController.data_repo_client.query_snapshot_indexes(query_json)
+      def self.get_tdr_results(selected_facets:, terms:)
         results = {}
-        raw_tdr_results.dig('result').each do |result_row|
-          short_name = result_row.dig('project_short_name')
+        if selected_facets.present?
+          facet_json = ApplicationController.data_repo_client.generate_query_from_facets(selected_facets)
+        end
+        if terms.present?
+          term_json = ApplicationController.data_repo_client.generate_query_from_keywords(terms)
+        end
+        # now we merge the two queries together to perform a single search request
+        query_json = ApplicationController.data_repo_client.merge_query_json(facet_query: facet_json, term_query: term_json)
+        logger.info "Executing TDR query with: #{query_json}"
+        raw_tdr_results = ApplicationController.data_repo_client.query_snapshot_indexes(query_json)
+        raw_tdr_results['result'].each do |result_row|
+          # get column name mappings for assembling results
```

Comment:
this section should get pulled out into a helper method `process_tdr_result_row`.  That will help in clarity and testing

Reply:
Good call.